src/test/lib/optimizer/strategy/chunk_pruning_rule_test.cpp

```diff
@@ -102,351 +95,386 @@ TEST_F(ChunkPruningRuleTest, SimplePruningTest) {
-  EXPECT_EQ(pruned_chunk_ids_table_2, expected_chunk_ids_table_2);
+  const auto expected_chunk_ids_table_2 = std::vector<ChunkID>{ChunkID{0}, ChunkID{1}, ChunkID{3}};
+  EXPECT_EQ(stored_table_node_2->pruned_chunk_ids(), expected_chunk_ids_table_2);
 }
 
 TEST_F(ChunkPruningRuleTest, ValueOutOfRange) {
   // Filters are not required to handle values out of their data type's range and the ColumnPruningRule currently
-  // doesn't convert out-of-range values into the type's range
+  // doesn't convert out-of-range values into the type's range.
   // TODO(anybody) In the test LQP below, the ChunkPruningRule could convert the -3'000'000'000 to MIN_INT (but ONLY
   //               as long as the predicate_condition is >= and not >).
 
-  auto stored_table_node = std::make_shared<StoredTableNode>("compressed");
+  const auto stored_table_node = StoredTableNode::make("compressed");
 
   // clang-format off
-  auto input_lqp =
+  const auto input_lqp =
   PredicateNode::make(greater_than_equals_(lqp_column_(stored_table_node, ColumnID{0}), int64_t{-3'000'000'000}),
     stored_table_node);
   // clang-format on
 
-  auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+  const auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
 
   EXPECT_EQ(actual_lqp, input_lqp);
-  std::vector<ChunkID> expected_chunk_ids = {};
-  std::vector<ChunkID> pruned_chunk_ids = stored_table_node->pruned_chunk_ids();
+  const auto expected_chunk_ids = std::vector<ChunkID>{};
+  EXPECT_EQ(stored_table_node->pruned_chunk_ids(), expected_chunk_ids);
+}
+
+// Predicates that have a subquery as an argument cannot be used for pruning as we do not know the predicate value yet.
+// However, such predicates should not prevent us from pruning by other predicates. The following two test cases ensure
+// correct pruning for predicates with uncorrelated and correlated subqueries.
+TEST_F(ChunkPruningRuleTest, PredicateWithUncorrelatedSubquery) {
+  const auto stored_table_node_1 = StoredTableNode::make("compressed");
+  const auto stored_table_node_2 = StoredTableNode::make("compressed");
+
+  const auto subquery_1 = ProjectionNode::make(expression_vector(value_(1.1)), DummyTableNode::make());
+  const auto subquery_2 = subquery_1->deep_copy();
+  const auto subquery_3 = ProjectionNode::make(expression_vector(value_(3.3)), DummyTableNode::make());
+
+  // clang-format off
+  // SELECT * FROM compressed WHERE a > 200 AND b = (SELECT 1.1);
+  const auto input_lqp_1 =
+  PredicateNode::make(greater_than_(stored_table_node_1->get_column("a"), value_(200)),  // Prunes chunk 1.
+    PredicateNode::make(equals_(stored_table_node_1->get_column("b"), lqp_subquery_(subquery_1)),
+      stored_table_node_1));
+
+  // SELECT * FROM compressed WHERE a > 200 AND b BETWEEN (SELECT 1.1) AND (SELECT 3.3);
+  const auto input_lqp_2 =
+  PredicateNode::make(greater_than_(stored_table_node_2->get_column("a"), value_(200)),  // Prunes chunk 1.
```

Comment:
Shouldn't matter, but can you pull the subquery-predicate up? Just to have both orders in this test.